### ISSUE TITLE
enhance LVGL port to SSD1306

### ISSUE THREAD

**Issue by dhebbeker** [dhebbeker/task-tracker#56]
The currently (7b84f5e69abdf7c3bccadfb38f9b38ea4694d5be) used [way to write pixels](https://github.com/Task-Tracker-Systems/Task-Tracker-Device/blob/7b84f5e69abdf7c3bccadfb38f9b38ea4694d5be/src/display.cpp#L37) is inefficient. As for each pixel the full display is re-written.

There are better approaches which would need adaptation:

- [old version (based on LVGL v6?) of lv_drivers, which contains an SSD1306 driver port](https://github.com/lvgl/lv_drivers/blob/bc24e9fd80db4746b3595245ba4a881265104f0d/display/SSD1306.c) which has been [abandoned](https://forum.lvgl.io/t/ssd1306-for-lv-port-esp32-available/1884/4).
- [a port to ESP32](https://github.com/lvgl/lvgl_esp32_drivers/blob/9fed1cc47b5a45fec6bae08b55d2147d3b50260c/lvgl_tft/ssd1306.c) which depends on ESP32-IDF (see also 525c765784f8d3116cecfa9b9b006aa0fb7f913c); abstraction of I2C interface and/or adaptation to Arduino framework would be necessary

Some alternative starting point: 82f4bf12a5e94ba4013daffe7575ca3bc43bbf29

Further resources:

- LVGL Forum: [Are there any (generic) display drivers ports for SSD1306 using I2C?](https://forum.lvgl.io/t/are-there-any-generic-display-drivers-ports-for-ssd1306-using-i2c/13352)
- Our wiki: [Display driver resources](https://github.com/Task-Tracker-Systems/Task-Tracker-Device/wiki/Display-driver-resources/20daaebcf0128ab6ecbe0ce1f37ae591db45f288)

💡 A successful port may be merged into [lv_drivers](https://github.com/lvgl/lv_drivers)
